### PR TITLE
feat: allows adjustable tab size in file editor

### LINKED ISSUE #1304: Allow adjustable tab size in file editor
### Requested feature:

Tab size is fixed to 2 spaces, making code difficult to read. Most editors allow changing tab size. This is a QOL improvement.

### Solves the following problem:

Currently tab size is 2 spaces, which makes code readability difficult for some files.

### Additional information:

Tab size adjustment can be added under Interface Settings -> Editor
image

### PR DESCRIPTION
**Pull request by forgodtosave**
With this PR, you can change the tab size in codemirror via the editor settings.

This PR will fix #1304

### CODE REVIEW

**Comment by meteyou** [comment]
hey! thx for your PR. Are you able to read/write every of this languages?

**Comment by forgodtosave** [comment]
Hi, not all of them. But de, ja, ru, hu should be correct and also nl and fr but here I'm not 100% sure.

**Comment by meteyou** [comment]
pls delete all translations, where you are not 100% sure. these should then be translated by native speakers.

**Comment by th33xitus** [comment]
Honestly i would prefer to remove the changes to ALL translations but german and (obviously) english. A lot of the other translations are outdated anyways and require an overall rework of a (at best) native speaker of those languages.

**Comment by forgodtosave** [comment]
I removed all translations except for en and de and languages where I asked a native speaker if it is correct.

**Comment by meteyou** [comment]
are you able to run prettier to fix the last code issues?

File: src/components/settings/SettingsEditorTab.vue [hunk 1]
```diff
@@ -67,6 +76,25 @@ export default class SettingsEditorTab extends Mixins(BaseMixin) {
         },
     ]
 
+    private tabSizes = [
```
Comment (meteyou) [inline src/components/settings/SettingsEditorTab.vue:79]: i would use here a getter method with a loop. pls also use i18n for the word "Spaces" here.

Comment (forgodtosave) [reply src/components/settings/SettingsEditorTab.vue]: Good idea, I have made a new commit with the suggested changes

File: src/components/settings/SettingsEditorTab.vue [hunk 2]
```diff
@@ -83,6 +100,14 @@ export default class SettingsEditorTab extends Mixins(BaseMixin) {
         this.$store.dispatch('gui/saveSetting', { name: 'editor.confirmUnsavedChanges', value: newVal })
     }
 
+    get tabSize() {
+        return this.$store.state.gui.editor.tabSize
```
Comment (meteyou) [inline src/components/settings/SettingsEditorTab.vue:104]: pls add here also a default value.

**Review by meteyou — APPROVED** [review]

**Comment by meteyou** [comment]
Thx for this PR!

**Pull request merged**